### ISSUE TITLE
library() + gnome.generate_gir() should force library being shared

### ISSUE THREAD

**Issue by ignatenkobrain** [mesonbuild/meson#851]
```
[brain@brain build]$ rm -rf * && meson .. && ninja-build  && ninja-build test
The Meson build system
Version: 0.34.0
Source dir: /home/brain/git/upstream/peas-test
Build dir: /home/brain/git/upstream/peas-test/build
Build type: native build
Build machine cpu family: x86_64
Build machine cpu: x86_64
Project name: libdnf
Native c compiler: cc (gcc 6.2.1)
Found pkg-config: /usr/bin/pkg-config (0.29.1)
Native dependency gobject-2.0 found: YES 2.50.0
Native dependency libpeas-1.0 found: YES 1.20.0
Build targets in project: 4
[7/7] 'Generating Dnf-1.0.typelib with a custom command.'
[0/1] Running all tests.
1/1 exe                                     OK       0.01 s
Full log written to meson-logs/testlog.txt.
```

```
[brain@brain build]$ rm -rf * && meson .. --default-library static && ninja-build  && ninja-build test
The Meson build system
Version: 0.34.0
Source dir: /home/brain/git/upstream/peas-test
Build dir: /home/brain/git/upstream/peas-test/build
Build type: native build
Build machine cpu family: x86_64
Build machine cpu: x86_64
Project name: libdnf
Native c compiler: cc (gcc 6.2.1)
Found pkg-config: /usr/bin/pkg-config (0.29.1)
Native dependency gobject-2.0 found: YES 2.50.0
Native dependency libpeas-1.0 found: YES 1.20.0

Meson encountered an error in file libdnf/meson.build, line 16, column 6:
Gir target must be an executable or shared library
```

**Comment by ocrete** [comment]
Any reason this can't just allow building gir files against a static library? I'm pretty sure it works with autotools, and the gir file can be used to create bindings for compiled languages.

**Comment by ocrete** [comment]
I just tested with autotools, and it just works with static libraries.

**Comment by ocrete** [comment]
Fixing this requires fixing both gobject-introspection & meson:

* Meson pull request #4476
* https://gitlab.gnome.org/GNOME/gobject-introspection/merge_requests/72

**Comment by Tachi107** [comment]
@ignatenkobrain I believe this was fixed by #4476, right?

**Comment by dcbaker** [comment]
Looks like both mrs landed, so I’m closing this
